validations/src/validations.rs

```diff
@@ -24,14 +29,40 @@ pub fn transaction_inputs_sum(
 ) -> Result<u64, failure::Error> {
     let mut total_value = 0;
 
-    for input in &tx.inputs {
-        let pointed_value = pool
-            .get(&input.output_pointer())
-            .ok_or_else(|| TransactionError::OutputNotFound {
-                output: input.output_pointer(),
-            })?
-            .value();
-        total_value += pointed_value;
+    match transaction_tag(tx) {
+        // Commit exception to calculate inputs
+        TransactionType::Commit => {
+            match &tx.inputs[0] {
+                Input::DataRequest(dr_input) => {
+                    // Get DataRequest information
+                    let dr_pointer = dr_input.output_pointer();
+                    let dr_output =
+                        pool.get(&dr_pointer)
+                            .ok_or(TransactionError::OutputNotFound {
+                                output: dr_pointer.clone(),
+                            })?;
+
+                    match dr_output {
+                        Output::DataRequest(dr_state) => {
+                            total_value = dr_state.value / u64::from(dr_state.witnesses);
+                        }
+                        _ => Err(TransactionError::InvalidCommitTransaction)?,
+                    }
+                }
+                _ => Err(TransactionError::InvalidCommitTransaction)?,
+            }
+        }
+        _ => {
+            for input in &tx.inputs {
+                let pointed_value = pool
+                    .get(&input.output_pointer())
+                    .ok_or_else(|| TransactionError::OutputNotFound {
+                        output: input.output_pointer(),
+                    })?
+                    .value();
+                total_value += pointed_value;
+            }
+        }
```

Comment:
_Suggestion_: the match expression is too big, an improvement could be to move into their own functions the logic of each match arm